app/src/org/commcare/views/widgets/StringWidget.java

```diff
@@ -43,9 +44,13 @@ public StringWidget(Context context, FormEntryPrompt prompt, boolean secret) {
 
     public StringWidget(Context context, FormEntryPrompt prompt, boolean secret, boolean inCompactGroup) {
         super(context, prompt, inCompactGroup);
-        mAnswer = (EditText)LayoutInflater.from(getContext()).inflate(getAnswerLayout(), this, false);
+        mAnswer = (ClearFocusEditText)LayoutInflater.from(getContext()).inflate(getAnswerLayout(), this, false);
         mAnswer.setTextSize(TypedValue.COMPLEX_UNIT_DIP, mAnswerFontSize);
         mAnswer.setOnClickListener(this);
+        mAnswer.setOnTouchListener((v, event) -> {
```

Comment:
what if the view gained focus due to other events like clicking next on Keyboard.